### PR TITLE
Properly fetch user balance for pools when wallet is connected

### PR DESCRIPTION
**Pull request by lazarmitic**

### CODE REVIEW

File: packages/tempus-client_v2/src/providers/getBalanceProvider.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import UserBalanceProvider, { UserBalanceProviderParams } from './balanceProvider';
+
+// Key - User wallet address
+// Value - Provider instance
+const userBalanceProviderMap = new Map<string, UserBalanceProvider>();
+const getUserBalanceProvider = (params: UserBalanceProviderParams): UserBalanceProvider => {
+  let userBalanceProvider = userBalanceProviderMap.get(params.userWalletAddress);
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/providers/getBalanceProvider.ts:7]: in `UserBalanceProvider` we call `this.destroy()` inside `init()`, so what's the reason to save a `UserBalanceProvider` instance?

File: packages/tempus-client_v2/src/providers/balanceProvider.tsx [hunk 2]
```diff
@@ -1,134 +1,106 @@
...
+      lpTokenService.balanceOf(this.userWalletAddress),
+      statisticsService.getRate(poolConfig.backingToken),
+    ]);
+
+    const exitEstimate = await statisticsService.estimateExitAndRedeem(
+      poolConfig.ammAddress,
+      lpTokenBalance,
+      principalsBalance,
+      yieldsBalance,
+      true,
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/providers/balanceProvider.tsx:83]: what's `true`?

File: packages/tempus-client_v2/src/providers/balanceProvider.tsx [hunk 3]
```diff
@@ -1,134 +1,106 @@
...
-          }),
-        );
-
-        subscriptions$.add(
-          stream$.subscribe((balances: BigNumber[]) => {
-            updateUserBalanceForPool(poolConfig.address, balances);
-          }),
-        );
-      }
+      this.tokenContracts.push(tpsContract, tysContract);
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/providers/balanceProvider.tsx:41]: you're not pushing `lpTokenContract`, is that intentional?

**Review by danieledamiani — CHANGES_REQUESTED** [review]
a few questions

File: packages/tempus-client_v2/src/providers/getBalanceProvider.ts (hunk 1 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/providers/getBalanceProvider.ts:7]: Are you suggesting to change it to static?

File: packages/tempus-client_v2/src/providers/balanceProvider.tsx (hunk 2 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/providers/balanceProvider.tsx:83]: I will add a variable.

File: packages/tempus-client_v2/src/providers/balanceProvider.tsx (hunk 3 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/providers/balanceProvider.tsx:41]: ah, good catch, I will fix it.

Comment (lazarmitic) [reply packages/tempus-client_v2/src/providers/balanceProvider.tsx]: fixed

File: packages/tempus-client_v2/src/providers/balanceProvider.tsx (hunk 2 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/providers/balanceProvider.tsx]: fixed

File: packages/tempus-client_v2/src/providers/getBalanceProvider.ts (hunk 1 above)
Comment (danieledamiani) [reply packages/tempus-client_v2/src/providers/getBalanceProvider.ts:7]: I'm saying that maybe we don't need to save the instance in `userBalanceProviderMap` at all

Comment (lazarmitic) [reply packages/tempus-client_v2/src/providers/getBalanceProvider.ts:7]: oh, yeah, maybe. Hmm, yeah, I think we can do that. I will try it

Comment (lazarmitic) [reply packages/tempus-client_v2/src/providers/getBalanceProvider.ts]: fixed

File: packages/tempus-client_v2/src/providers/getBalanceProvider.ts [hunk 4]
```diff
@@ -2,16 +2,8 @@ import UserBalanceProvider, { UserBalanceProviderParams } from './balanceProvide
 
 // Key - User wallet address
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/providers/getBalanceProvider.ts:3]: and these comments?

**Review by danieledamiani — DISMISSED** [review]

File: packages/tempus-client_v2/src/providers/getBalanceProvider.ts (hunk 4 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/providers/getBalanceProvider.ts:3]: ahh, missed those, I will remove them

**Review by danieledamiani — APPROVED** [review]

**Pull request merged**